akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbJournal.scala

```diff
@@ -153,4 +158,25 @@ object SharedLeveldbJournal {
    */
   def setStore(store: ActorRef, system: ActorSystem): Unit =
     Persistence(system).journalFor(null) ! AsyncWriteProxy.SetStore(store)
+
+  /**
+   * Configuration to enable `TestJavaSerializer` in `akka-testkit` for
+   * for the messages used by `SharedLeveldbJournal`.
+   *
+   * For testing only.
+   */
+  def configToEnableJavaSerializationForTest: Config = {
+    ConfigFactory.parseString(s"""
+    akka.actor.serialization-bindings {
+      "akka.persistence.journal.AsyncWriteTarget$$WriteMessages" = java-test
```

Comment:
are these messages ever serialized? WOuldn't journals serialize the thing inside?

Reply:
it's when using the journal proxy: `SharedLeveldbJournal`